platform/macos/src/MGLMapView.mm

```diff
@@ -1290,7 +1290,19 @@ + (NSSet *)keyPathsForValuesAffectingVisibleCoordinateBounds {
 }
 
 - (MGLCoordinateBounds)visibleCoordinateBounds {
-    return [self convertRect:self.bounds toCoordinateBoundsFromView:self];
+    return [self convertRect:[self NSEdgeInsetsInsetRect:self.bounds edgeInsets:self.contentInsets] toCoordinateBoundsFromView:self];
+}
+
+- (CGRect)NSEdgeInsetsInsetRect:(CGRect)rect edgeInsets:(NSEdgeInsets)insets {
```

Comment:
This method doesn’t depend on `self`, so it can be a standalone C function (like `UIEdgeInsetsInsetRect()`) rather than an Objective-C method, as long as it’s defined before the line that calls it:

```objc
static CGRect MGLEdgeInsetsInsetRect(CGRect rect, NSEdgeInsets insets) {
```

(The MGL prefix ensures that the function doesn’t collide with any `NSEdgeInsetsInsetRect()` that Apple may provide in a future version of AppKit.)

Reply:
Thanks @1ec5